Add TSV class for tab-separated values

jsxs0 opened the pull request:
This PR adds a new TSV class to provide first-class support for tab-separated files:

1. Inherits from CSV class
2. Uses tab as default separator
3. Full interface parity with CSV
4. All class methods supported (parse, read, generate etc.)
5. Allows override of default separator if needed

---
Example usage:

```
# Use TSV class with default tab separator
TSV.read("data.tsv")
TSV.parse("a\tb\tc")

# Override separator if needed
TSV.read("data.csv", col_sep: ",")
```

---
The change is motivated by:

1. Providing better support for TSV files
2. Making tab-separated file handling more convenient

---
Inspired by https://github.com/ruby/csv/issues/272

kou commented:
Hmm. Do we need to override all class methods? Can we just override `TSV#initialize`?

```diff
diff --git a/lib/csv.rb b/lib/csv.rb
index f6eb32a..8c71b4d 100644
--- a/lib/csv.rb
+++ b/lib/csv.rb
@@ -2979,6 +2979,12 @@ class CSV
   end
 end
 
+class TSV < CSV
+  def initialize(data, **options)
+    super(data, **({col_sep: "\t"}.merge(options)))
+  end
+end
+
 # Passes +args+ to CSV::instance.
 #
 #   CSV("CSV,data").read
```

BTW, is `CSV::TSV` intentional? (`::TSV` may be easy to use but it may break other library's API that defines `::TSV`...)

jsxs0 commented:
@kou Thanks for the feedback! Updated and pushed a new commit.

kou commented:
Could you add some tests for this change?
Could you update the PR description for the latest changes?

jsxs0 commented:
> Could you add some tests for this change? Could you update the PR description for the latest changes?

@kou Thanks, added tests and updated the description.

kou commented on run-test.rb:
Why do we need to change this file?
I think that we don't need to change this file but if you think that we need to change this file, could you open a separated PR?

kou commented on test/csv/test_tsv.rb:
```diff
@@ -0,0 +1,32 @@
+
+class TestTSV < Test::Unit::TestCase
+  def test_default_separator
+    tsv = TSV.new(String.new)
```
```suggestion
    tsv = CSV::TSV.new(String.new)
```

kou commented:
Did you run added tests on your local machine? Or did you enable GitHub Actions on your fork?
 
We need to add `CSV::` to all `TSV` references.

jsxs0 commented:
> Did you run added tests on your local machine? Or did you enable GitHub Actions on your fork?
> 
> We need to add `CSV::` to all `TSV` references.

Thanks. The fix was a straightforward namespace correction that aligns the test code with the actual implementation in `lib/csv.rb`.

kou commented:
Thanks.
